semtypes/src/main/java/io/ballerina/types/PredefinedType.java

```diff
@@ -244,27 +248,36 @@ static String toString(BasicTypeBitSet ut) {
         if ((ut.bitset & ERROR.bitset) != 0) {
             sb.add("error");
         }
-        if ((ut.bitset & LIST.bitset) != 0) {
-            sb.add("list");
-        }
-        if ((ut.bitset & FUNCTION.bitset) != 0) {
-            sb.add("function");
-        }
         if ((ut.bitset & TYPEDESC.bitset) != 0) {
             sb.add("typedesc");
         }
         if ((ut.bitset & HANDLE.bitset) != 0) {
             sb.add("handle");
         }
-        if ((ut.bitset & BasicTypeCode.VT_INHERENTLY_IMMUTABLE) != 0) { // TODO: fix when porting readonly
-            sb.add("readonly");
+        if ((ut.bitset & FUNCTION.bitset) != 0) {
+            sb.add("function");
+        }
```

Comment:
Wwould it be cleaner if we have a private method that take the "bitset" and string builder and the string if bit is set. So we can replace all these if checks with method calls.
```suggestion
       addIfBitSet(Function.bitset, "function");
```

Reply:
Updated.